pixl_imaging/tests/test_imaging_processing.py

```diff
@@ -171,6 +229,107 @@ async def test_querying_without_uid(orthanc_raw, caplog) -> None:
     assert await study.query_local(orthanc)
 
     expected_msg = (
-        f"No study found with UID {study.message.study_uid}, trying MRN and accession number"
+        f"No study found in modality UCVNAQR with UID {study.message.study_uid}, "
+        "trying MRN and accession number"
+    )
+    assert expected_msg in caplog.text
+
+
+class MockDateTime(datetime.datetime):
+    """Mock the date and time to be a Monday at 2 am."""
+
+    @classmethod
+    def now(cls, tz=None) -> datetime.datetime:
+        return MockDateTime(2024, 1, 1, 2, 0, tzinfo=tz)
+
+
+@pytest.mark.processing()
+@pytest.mark.asyncio()
+@pytest.mark.usefixtures("_add_image_to_fake_vna")
+@pytest.mark.usefixtures("_add_image_to_fake_pacs")
+async def test_querying_pacs_with_uid(orthanc_raw, caplog, monkeypatch) -> None:
+    """
+    Given a message with study_uid exists in PACS but not VNA,
+    When we query the archives
+    Then the querying finds the study in PACS with the study_uid
+    """
+    study = ImagingStudy.from_message(pacs_message)
+    orthanc = await orthanc_raw
+
+    assert not await study.query_local(orthanc)
+
+    with monkeypatch.context() as m:
+        # PACS is not queried during the daytime nor at the weekend.
+        # Set today to be a Monday at 2 am.
+        m.setattr(
+            datetime,
+            "datetime",
+            MockDateTime,
+        )
+        await process_message(pacs_message)
+
+    assert await study.query_local(orthanc)
+
+    expected_msg = (
+        f"No study found in modality UCVNAQR with UID {study.message.study_uid}, "
+        "trying MRN and accession number"
+    )
+    assert expected_msg in caplog.text
+
+    expected_msg = (
+        f"Failed to find study {study.message.study_uid} in primary archive, "
+        "trying secondary archive"
+    )
+    assert expected_msg in caplog.text
+
+    unexpected_msg = (
+        f"No study found in modality UCPACSQR with UID {study.message.study_uid}, "
+        "trying MRN and accession number"
+    )
+    assert unexpected_msg not in caplog.text
+
+
+@pytest.mark.processing()
+@pytest.mark.asyncio()
+@pytest.mark.usefixtures("_add_image_to_fake_vna")
+@pytest.mark.usefixtures("_add_image_to_fake_pacs")
+async def test_querying_pacs_without_uid(orthanc_raw, caplog, monkeypatch) -> None:
+    """
+    Given a message with non-existent study_uid exists in PACS but not VNA,
+    When we query the archives
```

Comment:
how difficult would it be to set up a test to see that if the secondary dicom isn't defined, then it doesn't try querying it?

Reply:
Yeah a test would be good, but I'm not sure how best to test it. I guess we could have a function that returns the list of modalities in orthanc, and mock it in the test to not return pacs?

Reply:
Could we make it look for the environmental variable within the function that runs, and then if its not defined at that time, then it doesn't use it? That way we could set the environmental variable for the setup of just that test I think